flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/AvailableNodesGenerator.java

```diff
@@ -20,28 +20,52 @@
 
 import com.google.gson.Gson;
 import com.google.gson.JsonArray;
+import io.ballerina.flowmodelgenerator.core.model.Category;
 import io.ballerina.flowmodelgenerator.core.model.FlowNode;
 
-import java.util.List;
-
 /**
  * Generates available nodes for a given position in the diagram.
  *
  * @since 1.4.0
  */
 public class AvailableNodesGenerator {
 
+    private final Category.Builder rootBuilder;
+
+    public AvailableNodesGenerator() {
+        this.rootBuilder = new Category.Builder(Category.Name.ROOT, null);
+        initializeCommonNodes();
+    }
+
     public JsonArray getAvailableNodes() {
         Gson gson = new Gson();
-        return gson.toJsonTree(
-                List.of(
-                        FlowNode.Kind.IF,
-                        FlowNode.Kind.HTTP_API_GET_CALL,
-                        FlowNode.Kind.HTTP_API_POST_CALL,
-                        FlowNode.Kind.RETURN,
-                        FlowNode.Kind.EXPRESSION,
-                        FlowNode.Kind.WHILE,
-                        FlowNode.Kind.ERROR_HANDLER
-                )).getAsJsonArray();
+        Category rootCategory = this.rootBuilder.build();
```

Comment:
Shall we cache this for now?

Reply:
Let's keep it as it is. In an ideal scenario, for each request, we have to derive this information. So no point of caching this.